src/libData/AccountData/AccountStoreSC.tpp

```diff
@@ -165,18 +160,40 @@ uint64_t AccountStoreSC<MAP>::InvokeEvmInterpreter(
     receipt.AddError(EXECUTE_CMD_TIMEOUT);
     ret = false;
   }
+}
+
+// New Invoker for EVM returns the gas, maybe change this
+template <class MAP>
+uint64_t AccountStoreSC<MAP>::InvokeEvmInterpreter(
+    Account* contractAccount, INVOKE_TYPE invoke_type,
+    EvmCallParameters& params, const uint32_t& version, bool& ret,
+    TransactionReceipt& receipt, evmproj::CallResponse& evmReturnValues) {
+  EvmCallRunner(invoke_type, params, version, ret, receipt, evmReturnValues);
+  uint64_t gas = params.m_available_gas;
   if (not evmReturnValues.isSuccess()) {
     LOG_GENERAL(WARNING, evmReturnValues.ExitReason());
   }
   // switch ret to reflect our overall success
   ret = evmReturnValues.isSuccess() ? ret : false;
 
-  if (evmReturnValues.Logs().size() > 0) {
+  if (!evmReturnValues.Logs().empty()) {
+    Json::Value _json = Json::arrayValue;
+    bool success = true;
```

Comment:
Not keen to block submission of this because of this comment: we need to move fast. Don't mind if the new style is being iterated over in parallel.

I feel this should also be discussed with other C++ style guideline questions. Sure it can be changed, if there is any merit.